### PR TITLE
fix: add subscribe to newsletter modal on home page

### PR DESCRIPTION
**Pull request by AdamPawlinski**
### Related Issues

closes [#EN-311](https://vsf.atlassian.net/browse/EN-311)

### Short Description of the PR

Add modal for subscribing newsletter on Home Page following the figma design. 

### Pull Request Checklist

- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.
- [ ] I have tested my Pull Request on production build and (to my knowledge) it works without any issues

- I tested manually my code and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme
- [ ] I have written test cases for my code

- [ ] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

### CODE REVIEW

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            I confirm subscription
+          </SfButton>
+        </form>
+        <SfHeading
+          description="You can unsubscribe at any time"
+          :level="3"
+        />
+        <SfScrollable :maxContentHeight="isMobile ? 'auto' : '3.75rem'" :class="{ 'is-open': !isHidden }">
+          <p class="modal__content">
+            After signing up for the newsletter, you will receive special offers and messages from VSF via email.
```
Comment (Baroshem) [inline packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:39]: Here also the translation. For the SfLink you can use https://kazupon.github.io/vue-i18n/guide/interpolation.html#basic-usage.

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            We will not sell or distribute your email to any third party at any time.
+            Please see our <SfLink link="https://www.vuestorefront.io/privacy-policy">Privacy Policy</SfLink>
+          </p>
+          <template #view-all>
+            <SfButton
+              class="sf-button--text sf-scrollable__view-all desktop-only"
+              @click="isHidden = !isHidden"
+            >
+              <span v-if="isHidden">show more</span>
+              <span v-else>hide</span>
```
Comment (Baroshem) [inline packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:49]:
```suggestion
              <span>{{ isHidden ? 'show more' : 'hide' }}</span>
```

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue (hunk 1 above)
Comment (Baroshem) [reply packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:39]: https://github.com/vuestorefront/vue-storefront/pull/5971/files#r648038706

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            label="Email address"
+            v-model="emailAddress"
+            class="modal__input"
+          />
+          <SfButton class="modal__button" type="submit">
+            I confirm subscription
+          </SfButton>
+        </form>
+        <SfHeading
+          description="You can unsubscribe at any time"
```
Comment (Baroshem) [inline packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:34]: $t - translation using i18n

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      <div>
+        <SfHeading
+          :level="3"
+          :title="$t('Subscribe to newsletter')"
+          class="modal__title desktop-only"
+        />
+        <form  @submit.prevent="$emit('email', emailAddress)">
+          <SfInput
+            type="email"
+            label="Email address"
```
Comment (Baroshem) [inline packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:25]: $t

**Review by Baroshem — CHANGES_REQUESTED** [review]

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue (hunk 1 above)
Comment (AdamPawlinski) [reply packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:39]: Thank for the tip :)

File: packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        <SfScrollable :maxContentHeight="isMobile ? 'auto' : '3.75rem'" :class="{ 'is-open': !isHidden }">
+          <i18n tag="p" class="modal__content" path="subscribeToNewsletterModalContent">
+            <SfLink link="https://www.vuestorefront.io/privacy-policy">{{ $t('Privacy Policy') }}</SfLink>
+          </i18n>
+          <template #view-all>
+            <SfButton
+              class="sf-button--text sf-scrollable__view-all desktop-only"
+              @click="isHidden = !isHidden"
+            >
+              <span v-if="isHidden">{{ $t('show more') }}</span>
```
[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply packages/core/nuxt-theme-module/theme/components/NewsletterModal.vue:46]: No it did, but I haven''t finished all fixes yet.

File: packages/core/nuxt-theme-module/theme/pages/Home.vue [hunk 6]
```diff
@@ -250,12 +267,22 @@ export default {
...
+      ],
+      emailAddressNotification: ''
     };
   },
   methods: {
     toggleWishlist(index) {
       this.products[index].isInWishlist = !this.products[index].isInWishlist;
+    },
+    handleNewsletterClick() {
+      const { toggleNewsletterModal } = useUiState();
```
[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply packages/core/nuxt-theme-module/theme/pages/Home.vue:279]: ok, thx.

File: packages/commercetools/theme/lang/en.js [hunk 7]
```diff
@@ -124,23 +136,35 @@ export default {
   'Continue to billing': 'Continue to billing',
   'Payment methods': 'Payment methods',
   'Shipping address': 'Shipping address',
-  'Subtotal': 'Subtotal',
-  'Shipping': 'Shipping',
+  Subtotal: 'Subtotal',
+  Shipping: 'Shipping',
   'Total price': 'Total price',
-  'Payment': 'Payment',
+  Payment: 'Payment',
```
[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply packages/commercetools/theme/lang/en.js]: Yes you are right. Changed that.

**Comment by AdamPawlinski** [comment]
I had to remove mobile observer that causes inifite loop error. Now it should work properly.

**Review by Baroshem — COMMENTED** [review]
Only one small proposition for better readability and follow Vue standards :)

File: packages/core/nuxt-theme-module/theme/pages/Home.vue [hunk 8]
```diff
@@ -78,7 +78,21 @@
...
+            @click="handleNewsletterClick"
+          >
+            {{ $t('Subscribe') }}
+          </SfButton>
+        </template>
+      </SfCallToAction>
+    </LazyHydrate>
+
+    <LazyHydrate when-visible>
+      <NewsletterModal @email="onSubscribe" />
```
[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply packages/core/nuxt-theme-module/theme/pages/Home.vue]: changed

**Review by Baroshem — APPROVED** [review]

**Review by alefbarbeli — APPROVED** [review]

**Pull request merged**